docs/defaults-detailed.rst

```diff
@@ -9,169 +11,640 @@ them.
+  the ``br0`` and ``br1`` bridge state to ``present`` because this interface
+  layout will try to remove them by default.
 
 ``auto``
-  Boolean variable (``True``/``False``) which determines if a given network
-  interface will be brought up automatically at system start. By default all
-  interfaces are configured to start automatically (``True``).
+  Optional, boolean. If ``True``, the network interface will be brought up by
+  the ``networking`` service at boot time, which might be not want you actually
+  want in the newer, :command:`systemd`-based hosts. By default it will be set
+  to ``False``. See also ``allow`` parameter.
 
 ``allow``
-  Name or list of names of "subsystems" which can enable the interface, for
-  example ``hotplug``. By default, if nothing is specified and interface is not
-  configured manually, ``allow-hotplug`` will be added automatically so that
-  the ``systemd`` ``ifup@.service`` unit can work correctly.
-
-``addresses``
-  List of IPv4 or IPv6 addresses in the ``host/prefix`` format which should be
-  configured on this interface. You can specify multiple IPv4 or IPv6
-  addresses, they will be filtered according to the interface "family" - this
-  allows you to keep all of the host IP addresses in one list.
-
-  The IP addresses will be configured in the order they are listed. If the interface is
-  configured as ``static``, the first IPv4 address will be the main one, the rest
-  will be added with numbered labels. When the interface is configured to get its
-  IP address via DHCP, all IPv4 addresses will be added with numbered labels.
-
-``gateway``
-  IP address of the gateway configured for this interface; only one gateway can
-  be configured at the moment.
-
-``label``
-  Specify a string which will be added to the interface name as a label.
-  Labeled interface configuration is stored in separate configuration files in
-  :file:`/etc/network/interfaces.d/`.
+  Optional, boolean, string or YAML list. If set to ``False``, this option is
+  disabled. If ``True``, the hotplug subsystem can bring this interface up or
+  down when the hotplug event is detected. You can also specify a list of
+  specific conditions at which the interface is brought up, currently
+  recognized conditions are:
+
+  - ``auto``: bring the interface up at boot time by the ``networking``
+    service. This might not be what you want on newer systems.
+
+  - ``boot``: bring the interface up at boot time by ``ifup@.service``
+    :command:`systemd` unit. This will put any processes related to a given
+    interface in their separate cgroup, which allows for better control over
+    the network interface. This is a custom implementation of the ``auto``
+    mechanism managed by this Ansible role.
+
+  - ``hotplug``: bring the interface up/down at hotplug events. This condition
+    is required to be present for the ``ifup@.service`` :command:`systemd` unit
+    to work properly.
```

Comment:
Should be mentioned in the docs. You don’t need to write it in the comments :wink: